src/soroban/server.ts

```diff
@@ -235,6 +235,82 @@ export class Server {
     );
   }
 
+  /**
+   * Retrieves the WASM bytecode for a given contract.
+   *
+   * This method allows you to fetch the WASM bytecode associated with a contract
+   * deployed on the Soroban network. The WASM bytecode represents the executable
+   * code of the contract.
+   *
+   * @param {string|Address|Contract} contract    the contract ID containing the
+   *    WASM bytecode to retrieve, as a strkey (`C...` form), a {@link Contract},
+   *    or an {@link Address} instance
+   *
+   * @returns {Promise<Buffer>}   a Buffer containing the WASM bytecode
+   *
+   * @throws {Error} If the contract or its associated WASM bytecode cannot be
+   *    found on the network.
+   *
+   * @example
+   * const contractId = "CCJZ5DGASBWQXR5MPFCJXMBI333XE5U3FSJTNQU7RIKE3P5GN2K2WYD5";
+   * server.getContractWasm(contractId).then(wasmBuffer => {
+   *   console.log("WASM bytecode length:", wasmBuffer.length);
+   *   // ... do something with the WASM bytecode ...
+   * }).catch(err => {
+   *   console.error("Error fetching WASM bytecode:", err);
+   * });
+   */
+  public async getContractWasm(
```

Comment:
yoooooooooooooo :fire: :fire: :fire: good stuff :clap:

Reply:
but now you need to add tests to `/test/unit/server/soroban` :p 

if you don't want to (totally fine, it's out of scope here), you can just make this a private helper, instead, and I'll do the piping to add it to `Server` as part of #961.

Reply:
No problem Ill add a test.

Reply:
Well that took longer than I expected but it was a good learning experience!

Reply:
I warned ya ;) but thank you! The tests look thorough, great work :clap: